### PR TITLE
Move to v3 publishing

### PR DESCRIPTION
**Pull request by adiaaida**

### CODE REVIEW

**Comment by adiaaida** [comment]
@mmitche @epananth PTAL

**Review by tmat — APPROVED** [review]

**Comment by tmat** [comment]
Thanks!

**Comment by tmat** [comment]
@adiaaida @mmitche Should https://github.com/dotnet/sourcelink/pull/729 also have been updated with the latest Arcade version? Seems like it hasn't been updated in a week.

**Comment by adiaaida** [comment]
I don't know if we have promoted arcade since that was last updated. It looks like it has the same arcade sdk version as what is currently in arcade itself: https://github.com/dotnet/arcade/blob/main/global.json#L6

**Comment by adiaaida** [comment]
Also, I don't have merge rights, so whenever tests complete, @tmat would you mind completing this PR for me?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -153,6 +153,7 @@ stages:
 - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
   - template: eng\common\templates\post-build\post-build.yml
     parameters:
+      publishingInfraVersion: 3
```
Comment (danmoseley) [inline azure-pipelines.yml:156]: Just noticed the issue mentioned in the line below was resolved 2 years ago. @JohnTortugo 
https://github.com/dotnet/arcade/issues/2871

**Pull request merged**